core/storage/btree.rs

```diff
@@ -1852,6 +1907,36 @@ impl BTreeCursor {
         cell_idx
     }
 
+    fn find_index_cell(&self, page: &PageContent, key_bytes: &[u8]) -> usize {
+        let mut cell_idx = 0;
+        let cell_count = page.cell_count();
+        while cell_idx < cell_count {
+            match page
+                .cell_get(
+                    cell_idx,
+                    payload_overflow_threshold_max(page.page_type(), self.usable_space() as u16),
+                    payload_overflow_threshold_min(page.page_type(), self.usable_space() as u16),
+                    self.usable_space(),
+                )
+                .unwrap()
+            {
+                BTreeCell::IndexLeafCell(IndexLeafCell { payload, .. }) => {
+                    if key_bytes <= payload {
+                        break;
+                    }
+                }
+                BTreeCell::IndexInteriorCell(IndexInteriorCell { payload, .. }) => {
+                    if key_bytes <= payload {
+                        break;
+                    }
+                }
+                _ => todo!(),
```

Comment:
this should panic with a message saying we reached the wrong type of btreecell (and print the cell type ofc)